fuzz/fuzz_targets/coprocessor/codec/mysql/decimal.rs

```diff
@@ -0,0 +1,80 @@
+#![no_main]
+#[macro_use]
+extern crate libfuzzer_sys;
+extern crate tikv;
+
+use std::mem;
+use tikv::coprocessor::codec::mysql::decimal::{Decimal, RoundMode};
+
+fn make_u64<I>(iter: &mut I) -> u64
+where
+    I: Iterator<Item = u8>,
+{
+    let mut bytes = [0u8; 8];
+    for byte in bytes.iter_mut() {
+        *byte = iter.next().unwrap();
+    }
+    unsafe { mem::transmute(bytes) }
+}
+
+fn fuzz<I>(lhs: &Decimal, rhs: &Decimal, iter: &mut I)
+where
+    I: Iterator<Item = u8>,
+{
+    let _ = lhs.clone().abs();
+    let _ = lhs.ceil();
+    let _ = lhs.floor();
+    let _ = lhs.prec_and_frac();
+
+    let mode = match iter.next().unwrap() % 3 {
+        0 => RoundMode::HalfEven,
+        1 => RoundMode::Truncate,
+        _ => RoundMode::Ceiling,
+    };
+    let frac = iter.next().unwrap() as i8;
+    let _ = lhs.clone().round(frac, mode.clone());
+
+    let word_buf_len = iter.next().unwrap();
+    let _ = lhs.clone()
+        .round_with_word_buf_len(frac, word_buf_len, mode);
```

Comment:
Here we should call `round` instead of `round_with_word_buf_len`, or make sure the word_buf is less or equal than 9. Maybe we could make the function `round_with_word_buf_len `   in another PR.

Reply:
Ok, I will skip it.